src/spikeinterface/core/basesorting.py

```diff
@@ -162,21 +187,113 @@ def get_unit_spike_train(
             ).astype("int64")
 
         if return_times:
-            if self.has_recording():
-                times = self.get_times(segment_index=segment_index)
-                return times[spike_frames]
-            else:
-                segment = self._sorting_segments[segment_index]
-                t_start = segment._t_start if segment._t_start is not None else 0
-                spike_times = spike_frames / self.get_sampling_frequency()
-                return t_start + spike_times
+            start_time = (
+                self.sample_index_to_time(start_frame, segment_index=segment_index) if start_frame is not None else None
+            )
+            end_time = (
+                self.sample_index_to_time(end_frame, segment_index=segment_index) if end_frame is not None else None
+            )
+
+            return self.get_unit_spike_train_in_seconds(
```

Comment:
In the new design, if we do return_times=True then spikes are read twice.
line 186 and then here lin 197.
Am I wrong.
We should reorganize the if else logic.

Reply:
Yes, you are right, I changed this.